README.md

```diff
@@ -317,13 +382,14 @@ const className = css({
 ### Don't: programmatically generate CSS
 
 This system will not de-duplicate any CSS styles. If the contents of the CSS
-object produce a unique combination style then that will be added to the page.
+object produce a unique combination styles then that will be added to the page.
+
+Be wary of calling `css()` programmatically or wrapping it in a function that
+accepts parameters and generates CSS depending on those parameters.
 
-In general be wary of calling `css()` programmatically or wrapping it in
-a function that accepts parameters and generates CSS depending on those
-parameters. That pattern is fine if done sparingly -- for example, once as the
-app is first loading to initialize CSS values for a site-wide theme. But if
-done frequently it can lead to memory concerns.
+This pattern is fine if done sparingly -- for example, once or twice to
+initialize CSS values for a site-wide theme or themes. But if done frequently
+it can easily lead to thousands or more of style sheets in the page head.
 
```

Comment:
These are the 2 ways we are using the `css` function currently. Does it matter which way we use it, or should we say that these are the acceptable uses?

```js
const MyComponent = () => {
  const tokens = useTokens()
  const styles = getStyles(tokens)

  return (
    <div className={css(styles.wrapper)}>
      <h1 className={styles.title}>Some title</h1>
      <button className={styles.button}>Click Me</button>
    </div>
  )
}

const getStyles = tokens => {
  button: css({
    padding: tokens.Spacing.ButtonPrimary,
    '&:hover': {
      backgroundColor: tokens.BackgroundColor. ButtonPrimaryHover,
    }
  },
  title: {
    fontSize: tokens.FontSize.H1,
  },
  wrapper: {
    display: 'flex',
    alignItems: 'center',
    '@media(min-width: 300px)': {
      flexDirection: 'column',
    },
  },
}
```

Reply:
This is totally fine. As long as future maintainers don't get lost in trying to see where the post-processed `className` string lives and where the pre-processed object-'o-styles lives there are no functional problems with this approach. This approach even has the benefit of reusing existing styles across different components since each block is smaller and more focused there's a greater likelihood that the same block exists in another component.